exercise_013_clustered_sudoku_solver/src/main/scala/org/neopixel/SudokuIO.scala

```diff
@@ -54,20 +56,43 @@ object SudokuIO {
   import java.io.{BufferedReader, File, FileReader}
 
   import scala.language.postfixOps
-  class FileLineTraversable(file: File) extends Traversable[String] {
-    override def foreach[U](f: String => U): Unit = {
-      val input = new BufferedReader(new FileReader(file))
-      try {
-        var line = input.readLine
-        while (line != null) {
-          f(line)
-          line = input.readLine
+  class FileLineTraversable(file: File) extends Iterable[String] {
```

Comment:
why does this have to be traversable? or why do we need to write our own traversable logic?